### ISSUE TITLE
[Question] Trying to connect with Redis Server

### ISSUE THREAD

**Issue by akash159chauhan** [zzzprojects/EntityFramework-Plus#752]
Hi, 
I am trying to use your FromCache() Extention and using Redis Cache as caching provider. I have configured the QueryCacheManager.Cache like this
`var connectionMultiplexer = ConnectionMultiplexer.Connect("localhost:6379");			
QueryCacheManager.Cache = new RedisObjectCache(connectionMultiplexer.GetDatabase(), Newtonsoft.Json.JsonConvert.SerializeObject, (type, s) => Newtonsoft.Json.JsonConvert.DeserializeObject(s, type));`

And when I am using your default CacheKeyFactory then it is throwing me below error
![image](https://user-images.githubusercontent.com/101166464/216272651-a49352fd-ac18-4f6a-a8a3-e62708c5939e.png)

And when I configure the CacheKeyFactory to 
`QueryCacheManager.CacheKeyFactory = (context, key) =>
{
// Custom cache key generation logic
var cacheKey = $"{key}_{context}";
return cacheKey;
};`

then it throws me this error
![image](https://user-images.githubusercontent.com/101166464/216273387-b18f74a7-e0dc-48a7-91bd-d9ff9bd651be.png)

And this is how I am using your FromCache() Extention
`IQueryable<User> users = _context.User;

var service = _context.User.FromCache("A1").ToList();`

Basically I have tested your FromCache() Extention with IMemoryCache everything as default that time it works fine for me, then I have a requirement where I need to support Distributed Caching then I got to know this library supports Redis which is used for distributed caching.

**Comment by JonathanMagnan** [comment]
Hello @akash159chauhan ,

Do you think you could create a runnable project with the issue? It doesn’t need to be your project, just a new solution with the minimum code to reproduce the issue.

The current issue is to early and happens on the `GetObjectQuery`, which normally is only related to the LINQ query, so we are currently unsure how to reproduce this issue.

Best Regards,

Jon

---
**Sponsorship**
[Help us improve this library](https://github.com/sponsors/zzzprojects)

**Performance Libraries**
`context.BulkInsert(list, options => options.BatchSize = 1000);  `
[Entity Framework Extensions](https://entityframework-extensions.net/) • [Dapper Plus](https://dapper-plus.net/)

**Runtime Evaluation**
`Eval.Execute("x + y", new {x = 1, y = 2}); // return 3`
[C# Eval Function](https://eval-expression.net/)

**Comment by akash159chauhan** [comment]
Hi Jon, this is the link of the Project https://github.com/akash159chauhan/WebApplication you can check the UserController where I am testing your Library, and also suggest if I am doing something wrong as I have very small knowledge of using Cache.

**Comment by JonathanMagnan** [comment]
Hello @akash159chauhan ,

In your project, you are currently using a package that is for `EF6` and not `EF Core 6`. You need to use the following package and match the EF Core major version, which the one from this package (in this case, `v6.x): https://www.nuget.org/packages/Z.EntityFramework.Plus.EFCore/6.18.3

Unfortunately, after looking more, it looks like we still doesn't support `Redis` for EF Core. We already have more tasks than we currently handle in the short-term plan, so unfortunately, adding support to `Redis` is not something we can currently do.

**Comment by akash159chauhan** [comment]
Hi Jon,
Yes I know that you have not given support of Redis for EF Core but it is supported in EF6 package, and I have tested with the package version which you have suggested but the problem is still same.
Aand I want to know if I create a custom class and Inherit with ObjectCache, to give the similar support of Redis like you have created RedisObjectCache then can FromCache() extentions will support this custom class out of the box, or I also have to create similar FromCache() extention to support Redis. As my need is that I want to support two caching functionality:
1. First which is going to use IMemoryCache for non-distributed environment 
2. Second which is going to use Redis for Distributed environment.
And on the basis of Cache Configuration the FromCache() extension will going to switch between the above two cache.

**Comment by JonathanMagnan** [comment]
Hello @akash159chauhan ,

The only way to know your answer is to try it on your side. Unfortunately, as we do not support it yet, we don't have the answer yet either to your question.

**Comment by akash159chauhan** [comment]
Hi Jon, did you have any running working example of using RedisObjectCache(). And Example or Unit Test can do a work for me. Thanks.

**Comment by JonathanMagnan** [comment]
It has been a long time since we played with `Redis` but we have a lab that look like this:

```csharp
public string SerializeObject(object objToSerialize)
{
	return Newtonsoft.Json.JsonConvert.SerializeObject(objToSerialize);
}

public object DeserializeObject(Type entiType, string serializedString)
{
	return Newtonsoft.Json.JsonConvert.DeserializeObject(serializedString, entiType);
}

var lazyConnection = new Lazy<ConnectionMultiplexer>(() =>
{

	string cacheConnection = "127.0.0.1:6379,allowAdmin=true"; 
	return ConnectionMultiplexer.Connect(cacheConnection);
});

QueryCacheManager.Cache = new RedisObjectCache(lazyConnection.Value.GetDatabase(), SerializeObject, DeserializeObject);
```
